### ISSUE TITLE
The directory src needs a major rework

### ISSUE THREAD

**Issue by KaruroChori** [BespokeSynth/BespokeSynth#345]
Having all that code in one single directory without a logical division among component types makes for a very challenging task to maintain the code in my opinion. It is hard to identify where is the entry point, which are the utility classes, etc.
I would suggest to design a new layout for the folders, so that we can make this project more scalable and more easily accessible for future developments.

**Comment by baconpaul** [comment]
yeah we've been doing that in the last couple of weeks with the libraries and awwbees and i have chatted about a restructure after 1.0.1 to make a core and modules and app more clearly divided for a few reasons. Happy to chat in the discord if you are on the server!

**Comment by KaruroChori** [comment]
:+1: 
Yes, I'm there as Chara, feel free to contact me.
Still, I find easier to track things on github; discord can be a bit overwhelming for my standards :smile: .

**Comment by baconpaul** [comment]
Yup. But anyway short version is "agreed; and there's a plan; but 1.0.1 comes first". Mostly wanted to make sure you didn't take a swing at it first :)

**Comment by KaruroChori** [comment]
Aye, that is why I asked before, as it seems to be a pretty laborious task and having a shared strategy on such architectural changes would be advisable :grin: . Also, since it is a major refactoring, I would expect this to break the merging process for any of the active forks, and the last thing one could want is to handle a fully manual revision.

By the way, I cannot find any plan for future releases in this repo, is there any kind of informal timeline for future developments? What should 1.0.1 entail?

**Comment by baconpaul** [comment]
maybe plan was generous. there's been some chats he and i have had basically which will help sort of complete the work we started with cmake and azure to get to nightlies and more organized source. 1.0.1 is mostly a big fix release and i think it is 'soon'.

**Comment by baconpaul** [comment]
oh and agree on forks, but git merges pretty well through moves (we have reorganized the surge codebase a few times) so i think we can make changes pretty safely once we turn to them.
